Dynamic-Programming/Abbreviation.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @description
+ * Given two strings, `a` and `b`, determine if it's possible to make `a` equal
+ * to `b` You can perform the following operations on the string `a`:
+ * 1. Capitalize zero or more of `a`'s lowercase letters.
+ * 2. Delete all the remaining lowercase letters in `a`.
+ *
+ * ### Algorithm
+ * The idea is in the problem statement itself: iterate through characters of
+ * string `a` and `b` (for character indexes `i` and `j` respectively):
+ * 1. If `a[i]` and `b[j]` are equal, then move to next position
+ * 2. If `a[i]` is lowercase of `b[j]`, then explore two possibilities:
+ * a) Capitalize `a[i]` or
+ * b) Skip `a[i]`
+ * 3. If the `a[i]` is not uppercase, just discard that character, else return
+ * `false`
+ *
+ * Time Complexity: (O(|a|*|b|)) where `|a|` => length of string `a`
+ *
+ * @param {String} a
+ * @param {String} b
+ * @returns {Boolean}
+ * @see https://www.hackerrank.com/challenges/abbr/problem - Related problem on HackerRank.
+ */
+export const abbreviation = (a, b) => {
```

Comment:
I'd probably call this `isAbbrevation`. Also, please give `a` and `b` more descriptive names like `word` and `abbr`.